cellmlmanip/model.py

```diff
@@ -619,6 +619,205 @@ def variables(self):
+
+    def _convert_state_variable_deriv(self, original_variable, new_variable, cf):
+        """
+        Create relevant variables/equations when converting a state variable.
+        :param original_variable: the variable to be converted
+        :param new_variable: the new variable representing the converted symbol
+        :param cf: conversion factor for unit conversion
+        :return: a dictionary containing the 'variable' and 'expression' for new derivative
+        """
+        # 1. find the derivative equation for this variable
+        # and get the free variable (wrt_variable)
+        eqn = None
+        for equation in self.equations:
+            if equation.args[0].is_Derivative:
+                if equation.args[0].args[0] == original_variable:
+                    eqn = equation
+                    break
+
+        # get free variable symbol
+        wrt_variable = eqn.args[0].args[1]
+
+        # 2. create new variable for derivative
+        new_deriv_variable = self._create_new_deriv_variable_and_equation(eqn, original_variable)
+
+        # 3. add a new derivative equation
+        expression = sympy.Eq(sympy.Derivative(new_variable, wrt_variable), new_deriv_variable * cf)
+        self.add_equation(expression)
+        return {'variable': new_deriv_variable, 'expression': eqn.args[0]}
+
+    def _convert_variable_instance(self, original_variable, cf, units):
+        """
+        Internal function to create new variable and an equation for it.
+        :param original_variable: VariableDummy object to be converted
+        :param cf: conversion factor
+        :param units: Unit object for new units
+        :return: the new variable created
+        """
+        # 1. get unique name for new variable
+        new_name = original_variable.name + '_converted'
+        while new_name in self.variables():
+            new_name = new_name + '_a'
+
+        # 2. if original has initial_value calculate new initial value
+        new_value = None
+        if original_variable.initial_value:
+            new_value = original_variable.initial_value * cf
+
+        # 3. copy cmeta_id from original and remove from original
+        new_variable = self.add_variable(name=new_name,
+                                         units=units,
+                                         initial_value=new_value,
+                                         cmeta_id=original_variable.cmeta_id)
+        original_variable.cmeta_id = ''
+
+        # 4. check whether we had an eqn for original
+        # if so, remove it and replace with new_var = rhs * cf
+        for equation in self.equations:
+            if equation.is_Equality and equation.args[0] == original_variable:
+                expression = sympy.Eq(new_variable, equation.args[1] * cf)
+                self.add_equation(expression)
+                self.remove_equation(equation)
+                break
+
+        # 5. add an equation for original variable
```

Comment:
Ah, so this is always using the 'input' flow, and so isn't suitable for 'output' cases, no?